pennylane/labs/resource_estimation/resource_container.py

```diff
@@ -0,0 +1,254 @@
+
+        Args:
+            op_type: the class object for an operation which inherits from '~.ResourceOperator'
+            params (dict): a dictionary containing the minimal pairs of parameter names and values
+                        required to compute the resources for the given operator!
+
+        .. details::
+
+            This representation is the minimal amount of information required to estimate resources for the operator.
+
+            **Example**
+
+            >>> op_tp = CompressedResourceOp(ResourceHadamard, {"num_wires":1})
+            >>> print(op_tp)
+            Hadamard(num_wires=1)
+        """
+        self._name = (op_type.__name__).replace("Resource", "")
+        self.op_type = op_type
+        self.params = params
+        self._hashable_params = tuple(params.items())
+
+    def __hash__(self) -> int:
+        return hash((self._name, self._hashable_params))
+
+    def __eq__(self, other: object) -> bool:
+        return (self.op_type == other.op_type) and (self.params == other.params)
+
+    def __repr__(self) -> str:
+        op_type_str = self._name + "("
+        params_str = ", ".join([f"{key}={self.params[key]}" for key in self.params]) + ")"
+
+        return op_type_str + params_str
+
+
+@dataclass
+class Resources:
+    r"""Contains attributes which store key resources such as number of gates, number of wires, and gate types.
+
+    Args:
+        num_wires (int): number of qubits
+        num_gates (int): number of gates
+        gate_types (dict): dictionary storing operation names (str) as keys
+            and the number of times they are used in the circuit (int) as values
+
+    .. details::
+
+        The resources being tracked can be accessed as class attributes.
+        Additionally, the :code:`Resources` instance can be nicely displayed in the console.
+
+        **Example**
+
+        >>> r = Resources(
+        ...             num_wires=2,
+        ...             num_gates=2,
+        ...             gate_types={"Hadamard": 1, "CNOT": 1}
+        ...     )
+        >>> print(r)
+        wires: 2
+        gates: 2
+        gate_types:
+        {'Hadamard': 1, 'CNOT': 1}
+    """
+
+    num_wires: int = 0
+    num_gates: int = 0
+    gate_types: defaultdict = field(default_factory=lambda: defaultdict(int))
+
+    def __add__(self, other: "Resources") -> "Resources":
+        """Add two resources objects in series"""
+        return add_in_series(self, other)
+
+    def __mul__(self, scalar: int) -> "Resources":
+        """Scale a resources object in series"""
+        return mul_in_series(self, scalar)
+
+    __rmul__ = __mul__  # same implementation
+
+    def __iadd__(self, other: "Resources") -> "Resources":
+        """Add two resources objects in series"""
+        return add_in_series(self, other, in_place=True)
+
+    def __imull__(self, scalar: int) -> "Resources":
+        """Scale a resources object in series"""
+        return mul_in_series(self, scalar, in_place=True)
```

Comment:
Adding/multiplying in series makes sense as the default - we will have to ensure documentation makes it clear that this is what the dunder methods are doing.